src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AppleCCCryptor.cs

```diff
@@ -0,0 +1,248 @@
+using System.Security.Cryptography;
+
+namespace Internal.Cryptography
+{
+    internal sealed class AppleCCCryptor : BasicSymmetricCipher
+    {
+        private readonly bool _encrypting;
+        private SafeAppleCryptorHandle _cryptor;
+
+        public AppleCCCryptor(
+            Interop.AppleCrypto.PAL_SymmetricAlgorithm algorithm,
+            CipherMode cipherMode,
+            int blockSizeInBytes,
+            byte[] key,
+            byte[] iv,
+            bool encrypting)
+            : base(cipherMode.GetCipherIv(iv), blockSizeInBytes)
+        {
+            _encrypting = encrypting;
+
+            OpenCryptor(algorithm, cipherMode, key);
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _cryptor?.Dispose();
+                _cryptor = null;
+            }
+
+            base.Dispose(disposing);
+        }
+
+        public override int Transform(byte[] input, int inputOffset, int count, byte[] output, int outputOffset)
+        {
+            Debug.Assert(input != null);
+            Debug.Assert(inputOffset >= 0);
+            Debug.Assert(count > 0);
+            Debug.Assert((count % BlockSizeInBytes) == 0);
+            Debug.Assert(input.Length - inputOffset >= count);
+            Debug.Assert(output != null);
+            Debug.Assert(outputOffset >= 0);
+            Debug.Assert(output.Length - outputOffset >= count);
+
+            return CipherUpdate(input, inputOffset, count, output, outputOffset);
+        }
+
+        public override byte[] TransformFinal(byte[] input, int inputOffset, int count)
+        {
+            Debug.Assert(input != null);
+            Debug.Assert(inputOffset >= 0);
+            Debug.Assert(count >= 0);
+            Debug.Assert((count % BlockSizeInBytes) == 0);
+            Debug.Assert(input.Length - inputOffset >= count);
+
+            byte[] output = ProcessFinalBlock(input, inputOffset, count);
+            Reset();
+            return output;
+        }
+
+        private unsafe byte[] ProcessFinalBlock(byte[] input, int inputOffset, int count)
+        {
+            if (count == 0)
+            {
+                return Array.Empty<byte>();
+            }
+
+            byte[] output = new byte[count];
+            int outputBytes = CipherUpdate(input, inputOffset, count, output, 0);
+            int ret;
+            int errorCode;
+
+            fixed (byte* outputStart = output)
+            {
+                byte* outputCurrent = outputStart + outputBytes;
+                int bytesWritten;
+
+                ret = Interop.AppleCrypto.CryptorFinal(
+                    _cryptor,
+                    outputCurrent,
+                    output.Length - outputBytes,
+                    out bytesWritten,
+                    out errorCode);
+
+                outputBytes += bytesWritten;
+            }
+
+            Exception exception = ProcessInteropError(ret, errorCode);
+
```

Comment:
Ugh, things touching an `if` is like nails on a blackboard to me.  I'll take the other suggestion and let the method do its own throwing :smile: